worker/provisioner/provisioner_task.go

```diff
@@ -421,7 +426,21 @@ func processOneMachineProfileChange(
 	m apiprovisioner.MachineProvisioner,
 	profileBroker environs.LXDProfiler,
 ) (bool, error) {
-	logger.Debugf("processOneMachineProfileChange(%s)", m.Id())
+	ident := m.Id()
+	logger.Debugf("processOneMachineProfileChange(%s)", ident)
+	if machineStatus, _, err := m.InstanceStatus(); err != nil {
+		return false, errors.Annotatef(err, "failed to get machine status %q", ident)
+	} else if machineStatus != status.Running {
+		if _, err := m.InstanceId(); err != nil && params.IsCodeNotProvisioned(err) {
+			logger.Tracef("Attempting to apply a profile to a machine that isn't provisioned %q", ident)
+			if err := m.RemoveUpgradeCharmProfileData(); err != nil {
+				logger.Tracef("cannot remove machine upgrade charm profile data: %s", err.Error())
+			}
+			// There is nothing we can do with this machine, we could continue
+			// on and let the uniter apply the profile at a later stage.
```

Comment:
The uniter does not apply profiles to a machine.  If the machine is not provisioned,  they'll be applied when the machine is provisioned.